amethyst_renderer/src/pass/flat2d/encode_image.rs

```diff
@@ -0,0 +1,101 @@
+use super::Flat2DData;
+use crate::{Flipped, Hidden, HiddenPropagate, Rgba, Texture, Transparent};
+use amethyst_assets::{AssetStorage, Handle};
+use amethyst_core::{
+    nalgebra::Vector4,
+    specs::{Component, Join, Read, ReadStorage, System, VecStorage, Write},
+    GlobalTransform,
+};
+
+/// Information for rendering a standalone texture as plain image.
+///
+/// Instead of using a `Mesh` on a `DrawFlat` render pass, we can use a simpler set of shaders to
+/// render textures to quads. This struct carries the information necessary for the `DrawFlat2D` pass.
+#[derive(Clone, Debug)]
+pub struct RenderImageFlat2D(pub Handle<Texture>);
```

Comment:
I'm not sure I like this naming of the struct, it seems a bit wonky/cumbersome. Perhaps `SpriteRender` for this (or `IndividualSpriteRender` if we really want to be explicit) and `SpriteSheetRender` for the version to be used with a SpriteSheet?

Reply:
There was some discussion about that before I made that into a PR. I understand your point and it is indeed visually more appealing, but the conclusion was that we currently have some ambiguity with existing type names. I wanted to make it as clear as possible that this type represents some kind of cross-section of `DrawFlat2D` and `Image`. Well, actually it could be `Texture` to make it perfectly consistent.

Reply:
It's important to note that other passes will have separate types even if you want to use the same shape of data with them. This is the only way for the user to specify the intent. L and to not influence another rendering pass by accident.